phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCQueryIT.java

```diff
@@ -135,6 +140,117 @@ private void cdcIndexShouldNotBeUsedForDataTableQueries(Connection conn, String
         String explainPlan = QueryUtil.getExplainPlan(rs);
         assertFalse(explainPlan.contains(cdcName));
     }
+
+    private boolean isDistinctPrefixFilterIncludedInFilterList(FilterList filterList) {
+        for (Filter filter : filterList.getFilters()) {
+            if (filter instanceof DistinctPrefixFilter) {
+                return true;
+            } else if (filter instanceof FilterList) {
+                return isDistinctPrefixFilterIncludedInFilterList((FilterList) filter);
+            }
+        }
+        return false;
+    }
+    private boolean isDistinctPrefixFilterIncluded(Scan scan) {
+        Filter filter = scan.getFilter();
+        if (filter != null && filter instanceof DistinctPrefixFilter) {
+            return true;
+        } else if (filter instanceof FilterList) {
+                return isDistinctPrefixFilterIncludedInFilterList((FilterList) filter);
+        }
+        return false;
+    }
+
+    private void checkIndexPartitionIdCount(Connection conn, String cdcName) throws Exception {
+        // Verify that we can use retrieve partition ids
+        ResultSet rs = conn.createStatement().executeQuery("SELECT PARTITION_ID() FROM "
```

Comment:
Add ORDER BY PARTITION_ID() to make sure rows are returned orderly here (by Tanuj)